azurerm/internal/services/network/tests/firewall_policy_resource_test.go

```diff
@@ -0,0 +1,261 @@
+		if err != nil {
+			return err
+		}
+
+		resp, err := client.Get(ctx, id.ResourceGroup, id.Name, "")
+		if err == nil {
+			return fmt.Errorf("Network.FirewallPolicies still exists")
+		}
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Getting on Network.FirewallPolicies: %+v", err)
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMFirewallPolicy_basic(data acceptance.TestData) string {
+	template := testAccAzureRMFirewallPolicy_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_firewall_policy" "test" {
+  name                = "acctest-fwpolicy-Policy-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMFirewallPolicy_complete(data acceptance.TestData) string {
+	template := testAccAzureRMFirewallPolicy_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_firewall_policy" "test" {
+  name                     = "acctest-fwpolicy-Policy-%d"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  threat_intelligence_mode = "Off"
+  threat_intelligence_whitelist {
+    ip_addresses = ["1.1.1.1", "2.2.2.2"]
+    fqdns        = ["foo.com", "bar.com"]
+  }
+  dns_setting {
+    servers                    = ["1.1.1.1", "2.2.2.2"]
+    proxy_enabled              = true
+    network_rules_fqdn_enabled = true
+  }
+  tags = {
+    env = "Test"
+  }
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMFirewallPolicy_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMFirewallPolicy_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_firewall_policy" "import" {
+  name                = azurerm_firewall_policy.test.name
+  resource_group_name = azurerm_firewall_policy.test.resource_group_name
+  location            = azurerm_firewall_policy.test.location
+}
+`, template)
+}
+
+func testAccAzureRMFirewallPolicy_inherit(data acceptance.TestData) string {
+	template := testAccAzureRMFirewallPolicy_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_firewall_policy" "test-parent" {
+  name                = "acctest-fwpolicy-Policy-%d-parent"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+
+resource "azurerm_firewall_policy" "test" {
+  name                = "acctest-fwpolicy-Policy-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  base_policy_id      = azurerm_firewall_policy.test-parent.id
+}
+`, template, data.RandomInteger, data.RandomInteger)
+}
+
+func testAccAzureRMFirewallPolicy_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-fwpolicy-%d"
```

Comment:
could we make this just the service? or sub-servie?
```suggestion
  name     = "acctestRG-networkfw-%d"
```

Reply:
Agreed!